src/client/puzzle/setup-puzzle.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { SetupBase } from "../setup/setup-base";
+import { SelectPuzzle } from "./select-puzzle";
+import { NonIdealState, Spinner } from "@blueprintjs/core";
+import { get, useEffectQuery } from "../api";
+import { Navigate } from "react-router-dom";
+
+export function SetupPuzzle() {
+    const [selectedPuzzle, setSelectedPuzzle] = useState<string | undefined>();
+
+    //get puzzles from api
+    const { isPending, data, isError } = useEffectQuery(
+        "get-puzzles",
+        async () => {
+            return get("/get-puzzles").then((puzzles) => {
+                return puzzles;
+            });
+        },
+        false,
+    );
+
+    if (isPending) {
+        return (
+            <NonIdealState
+                icon={<Spinner intent="primary" />}
+                title="Loading..."
+            />
+        );
+    } else if (isError) {
+        return <Navigate to="/home" />;
+    }
+
+    if (data === undefined) {
+        return <Spinner intent="primary" />;
```

Comment:
iirc if isPending and isError are both falsy, data should be filled with whatever value the endpoint returned. if it doesn't it should be counted as an error, probably the same way you handle the isError state